web/src/components/searchbox.tsx

```diff
@@ -36,7 +36,8 @@ const SearchBoxInternal: React.FC<{search: (
     disable: () => void,
     enable: (f_set: Followup[] | ((fs: Followup[]) => Followup[])) => void,
   ) => void,
-}> = ({search}) => {
+  onQuery?: (q: string) => any,
```

Comment:
adds a callback on changes to the search input. It's used to update the prompt preview